aws/elbv2_test.go

```diff
@@ -25,20 +25,36 @@ func createTestELBv2(t *testing.T, session *session.Session, name string) elbv2.
 		assert.Fail(t, "Needs at least 2 subnets to create ELBv2")
 	}
 
-	subnet1 := *subnetOutput.Subnets[0]
-	subnet2 := *subnetOutput.Subnets[1]
+	var firstSubnet ec2.Subnet
+	var secondSubnet ec2.Subnet
+
+	// ensure we select subnets in different availability zones
+	for _, subnet1 := range subnetOutput.Subnets {
+		for _, subnet2 := range subnetOutput.Subnets {
+			if subnet1.AvailabilityZone != subnet2.AvailabilityZone && subnet1.SubnetId != subnet2.SubnetId {
+				firstSubnet = *subnet1
+				secondSubnet = *subnet2
+				break
+			}
+
+			if subnet1 != nil && subnet2 != nil {
+				break
+			}
```

Comment:
I find the logic above a bit confusing. Couldn't you just set `subnet1 = *subnetOutput.Subnets[0]` and then just loop over `*subnetOutput.Subnets[1 ... n]` until you find one in a different AZ? Also, you have a check above for at least 2 subnets, but if they are in the same AZ, you may need more, so you probably need to check after this loop that two subnets were actually found!